flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -216,6 +216,9 @@ private void preValidate() {
 						+ "State checkpoints happen normally, but records in-transit during the snapshot will be lost upon failure. "
 						+ "\nThe user can force enable state checkpoints with the reduced guarantees by calling: env.enableCheckpointing(interval,true)");
 			}
+			if (streamGraph.isIterative() && !checkpointConfig.isForceUnalignedCheckpoints()) {
+				throw new UnsupportedOperationException("Unaligned Checkpoints are currently not supported for iterative jobs");
```

Comment:
I'd add more details similar to forcing checkpoints

> "Unaligned Checkpoints are currently not supported for iterative jobs, as Flink cannot rescale from these checkpoints. "
						+ "\nYou can force enable unaligned checkpoints if you don't intend to rescale: env.setForceUnalignedCheckpoints(true)"

Reply:
Done.